File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.java
class Solution {
    public int maxProduct(int n) {
        List<Integer> list = new ArrayList<>();
        int max =0;

        while(n>0){
            int d = n % 10;
            list.add(d);
            n = n/10;
        }
        for(int i =0; i<list.size(); i++){
            for(int j =i+1; j<list.size(); j++){
                max = Math.max(max, list.get(i) * list.get(j));

            }
        }
        return max;
        
    }
}

File: 3536-maximum-product-of-two-digits/3536-maximum-product-of-two-digits.js
var maxProduct = function (n) {
    let digits = [];
    let max = 0;

    while (n > 0) {
        digits.push(n % 10);
        n = Math.floor(n / 10);
    }
    for (let i = 0; i < digits.length; i++) {
        for (let j = i + 1; j < digits.length; j++) {
            max = Math.max(max, digits[i] * digits[j]);
        }
    }
    return max;
};